src/optimizer_core_interface.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 @artofnothingness Alexey Budyakov, Samsung Research
+
+#include "mppic/optimizer_core_interface.hpp"
+#include "nav2_costmap_2d/costmap_filters/filter_values.hpp"
+
+namespace mppi
+{
+
+
+void IOptimizerCore::setConstraints(double speed_limit, bool percentage)
```

Comment:
If its just a single function, you can do this in the header

Reply:
I'll probably remove IOptimizerCore and idea of pluginable optimizer